TopMovies/Core/Action/MoviesListAction.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  MoviesListAction.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 21.02.2021.
+//
+
+import ReSwift
+
+struct RequestedMoviesListAction: Action {
+  let categoryId: MovieCategory.ID
+  let requestType: RequestType
+  
+  enum RequestType { case reload, loadMore }
+}
+struct DownloadingMoviesListAction: Action {
+  let categoryId: MovieCategory.ID
+}
+struct CompletedMoviesListAction: Action {
+  let categoryId: MovieCategory.ID
```

Comment:
You need requestType in all actions too.
How would you know if that action has been triggered during reloading or loading more?
 
Inside of the state you check `if state.reload == .requested then state.reload = .downloading`.
But what if `CompletedMoviesListAction ` was triggered during the loadMore operation, or `DownloadingMoviesListAction` was lost and not triggered at all?

It's something that we could discuss, but I'd prefer to specify exactly what kind of request was started/completed/failed since handling that logic in the state may bring in many bugs.